src/native-stack/views/NativeStackView.tsx

```diff
@@ -403,19 +403,29 @@ function NativeStackViewInner({
 }: Props): JSX.Element {
   const { key, routes } = state;
 
+  const topScreenOptions = descriptors[state.routes[state.index].key].options;
```

Comment:
IMO I would extract `state.routes[state.index].key` or `state.routes[state.index]` to the separate variable just for readability reasons